### ISSUE TITLE
Internal - Deprecate usage of itest

### ISSUE THREAD

**Issue by dsherret** [denoland/deno#22907]
We'd like to deprecate the usage of itest and move the tests over to https://github.com/denoland/deno/tree/main/tests/specs

- [ ] bench_tests.rs
- [ ] bundle_tests.rs
- [ ] cache_tests.rs
- [ ] cert_tests.rs
- [ ] check_tests.rs
- [ ] compile_tests.rs
- [ ] coverage_tests.rs
- [ ] doc_tests.rs
- [ ] eval_tests.rs
- [ ] flags_tests.rs
- [ ] fmt_tests.rs
- [ ] info_tests.rs
- [ ] init_tests.rs
- [ ] inspector_tests.rs
- [ ] install_tests.rs
- [ ] js_unit_tests.rs
- [x] jsr_tests.rs
- [ ] jupyter_tests.rs
- [ ] lint_tests.rs
- [ ] lsp_tests.rs
- [ ] node_compat_tests.rs
- [ ] node_unit_tests.rs
- [ ] npm_tests.rs
- [ ] pm_tests.rs
- [ ] publish_tests.rs
- [ ] repl_tests.rs
- [ ] run_tests.rs
- [ ] shared_library_tests.rs
- [ ] task_tests.rs
- [ ] test_tests.rs
- [ ] upgrade_tests.rs
- [ ] vendor_tests.rs
- [ ] watcher_tests.rs
- [ ] worker_tests.rs

**Comment by dsherret** [comment]
No, by internal I just meant internal code. This would be a great thing for anyone to contribute! I'd really like to deprecate itest. If anyone works on this I'd recommend doing it a bit at a time so that the PRs are manageable.

**Comment by MohammadSu2000** [comment]
Can I pick this up?

**Comment by HasanAlrimawi** [comment]
I'm deprecating the integration tests of `eval_tests.rs` in favor of spec tests.

**Comment by bartlomieju** [comment]
@MohammadSu2000 @HasanAlrimawi please do!

**Comment by HasanAlrimawi** [comment]
I'm deprecating test_itests.rs

**Comment by MohammadSu1** [comment]
I'm deprecating the worker_tests.rs file

**Comment by MohammadSu1** [comment]
I'm deprecating the node_compat_tests.rs file

**Comment by HasanAlrimawi** [comment]
I'm deprecating task_tests.rs

**Comment by HasanAlrimawi** [comment]
I'm deprecating lint_tests.rs

**Comment by MohammadSu1** [comment]
I'm deprecating the npm_tests.rs

**Comment by ingalless** [comment]
@MohammadSu1 I was looking at the `check_tests` recently for another issue I was investigating. I'll deprecate `itest` in there if that's okay with you?

**Comment by majiayu000** [comment]
Hi, I've been looking into this and traced the root cause.

**Proposed approach:** 1) Clone denoland/deno and grep for remaining `itest!` macro usages in tests/integration/ or similar directories. 2) Pick one unclaimed test file. 3) Convert each itest! invocation to a spec-style test (using #[tokio::test] or similar spec runner pattern already established in merged PRs). 4) Reference issue #22907 in the PR body. Look at merged PRs like #25382 or #25452 as templates for the conversion pattern.

I'll open a draft PR shortly. Happy to adjust based on your preference.

**Comment by bartlomieju** [comment]
This is now done
